RecoTracker/TransientTrackingRecHit/python/TTRHBuilders_cff.py

```diff
@@ -7,8 +7,6 @@
 from RecoTracker.TkSeedingLayers.TTRHBuilderWithoutAngle4PixelPairs_cfi import *
 from RecoTracker.TkSeedingLayers.TTRHBuilderWithoutAngle4PixelTriplets_cfi import *
 #TransientTRH builder with template
-from RecoLocalTracker.SiPixelRecHits.pixelCPEFastESProducerPhase1_cfi import *
-from RecoLocalTracker.SiPixelRecHits.pixelCPEFastESProducerPhase2_cfi import *
```

Comment:
when doing this, you might want to remove also this:

https://github.com/cms-sw/cmssw/blob/78b3f7441ecea1919648e50a149076c3588ab709/RecoTracker/TransientTrackingRecHit/python/TransientTrackingRecHitBuilder_cfi.py#L11-L12

I had a quick look on to replace this functionality with an `alpaka`-based CPE, but it's not entirely trivial.
`PixelCPEFastParamsESProducerAlpaka` produces data in a record called `PixelCPEFastParamsRecord`

https://github.com/cms-sw/cmssw/blob/78b3f7441ecea1919648e50a149076c3588ab709/RecoLocalTracker/SiPixelRecHits/plugins/alpaka/PixelCPEFastParamsESProducerAlpaka.cc#L31

(which is what the [HLT chain for pixel rechits and pixel tracks uses](https://github.com/search?q=repo%3Acms-sw%2Fcmssw%20PixelCPEFastParamsRecord&type=code)) and not `TkPixelCPERecord` which is the record that the `TransientTrackingRechitBuilder` is expecting to be filled:

https://github.com/cms-sw/cmssw/blob/78b3f7441ecea1919648e50a149076c3588ab709/RecoTracker/TransientTrackingRecHit/plugins/TkTransientTrackingRecHitBuilderESProducer.cc#L41

Notably the `PixelCPEFastESProducer` instead did produce data in `TkPixelCPERecord`: 

https://github.com/cms-sw/cmssw/blob/78b3f7441ecea1919648e50a149076c3588ab709/RecoLocalTracker/SiPixelRecHits/plugins/PixelCPEFastESProducer.cc#L58-L59

which is why the CUDA code could and can still run without problems.
@ferencek @mroguljic FYI

Reply:
Thanks for looking into this.
So the question is, do we want to keep the possibility of using the FastCPE in the non-alpaka code ?

Reply:
> we want to keep the possibility of using the FastCPE in the non-alpaka code ?

I (personally) don't find it particularly useful. The motivation to have the fast CPE available in the track refitting was to use it to evaluate the pixel local reconstruction performance in slightly more realistic "HLT conditions" with the existing pixel offline and tracker alignment tools. Once the CUDA code has ended being actively used in the HLT, this doesn't serve much purpose. 
IMHO supporting the alpaka code for the same reasons (and perhaps toward using it for phase2 in full offline tracking?) might be a better time investment.

Reply:
However, we now ended up in a situation where we wanted to validate the performance of the Irradiation Bias Correction (IBC) implemented in the Alpaka-based code and hit a wall because of the above-mentioned differences in the ESProducers which do not allow testing the Alpaka-based implementation in the current track refitting code. What I also wasn't aware of was that even at the HLT for track fitting the PixelCPEGeneric is being used, where the IBC is already enabled, so in a way the whole effort of having the IBC in the Alpaka-based code is not that critical (perhaps not even needed).

Reply:
> , so in a way the whole effort of having the IBC in the Alpaka-based code is not that critical (perhaps not even needed).

Well, arguably having a bias correction in the rechit building used in the track seeding should be more important than the CPE used to fit the final tracks (tracking efficiency vs parameter resolution).

Reply:
According to the DQM plots, it appears that the IBC at the level of rechit building slightly worsens the rechit resolution. Only at the level of transient tracking rechits with a clear track hypothesis defined one starts to see improvements.

Reply:
I've opened this https://github.com/cms-sw/cmssw/pull/46852 to realign the `PixelCPEFastParams` product with the other CPEs.

Reply:
(a draft for the moment, given the subsequent follow up with @mmusich about using it in the refitter)